components/perf-tools/geopm/SPECS/geopm.spec

```diff
@@ -0,0 +1,570 @@
+libgeopm library, the libgeopmpolicy library, the geopmctl application and the
+geopmpolicy application.  The libgeopm library can be called within MPI
+applications to enable application feedback for informing the control
+decisions.  If modification of the target application is not desired then the
+geopmctl application can be run concurrently with the target application.  In
+this case, target application feedback is inferred by querying the hardware
+through Model Specific Registers (MSRs).  With either method (libgeopm or
+geopmctl), the control hierarchy tree writes processor power policy through
+MSRs to enact policy decisions.  The libgeopmpolicy library is used by a
+resource manager to set energy policy control parameters for MPI jobs.  Some
+features of libgeopmpolicy are available through the geopmpolicy application
+including support for static control.
+
+%prep
+
+%setup -q -n %{pname}-%{version}
+
+%build
+%ohpc_setup_compiler
+test -f configure || ./autogen.sh
+./configure --prefix=%{install_path} \
+            --disable-fortran --disable-doc \
+            || ( cat config.log && false )
+%{__make}
+
+
+%install
+%ohpc_setup_compiler
+%{__make} DESTDIR=%{buildroot} install
+rm -f $(find %{buildroot}/%{install_path} -name '*.a'; \
+        find %{buildroot}/%{install_path} -name '*.la'; \
+        find %{buildroot}/%{install_path} -name 'geopm_launcher.1*')
+
+# Module file
+# OpenHPC module file
+%{__mkdir_p} %{buildroot}%{OHPC_MODULEDEPS}/%{compiler_family}-%{mpi_family}/%{pname}
+%{__cat} << EOF > %{buildroot}/%{OHPC_MODULEDEPS}/%{compiler_family}-%{mpi_family}/%{pname}/%{version}
+#%Module1.0#####################################################################
+
+proc ModulesHelp { } {
+
+puts stderr " "
+puts stderr "This module loads the %{pname} library built with the %{compiler_family} compiler"
+puts stderr "toolchain and the %{mpi_family} MPI stack."
+puts stderr "\nVersion %{version}\n"
+
+}
+module-whatis "Name: %{pname} built with %{compiler_family} compiler and %{mpi_family} MPI"
+module-whatis "Version: %{version}"
+module-whatis "Category: runtime"
+module-whatis "Description: %{summary}"
+module-whatis "URL %{url}"
+
+set     version             %{version}
+
+prepend-path    PATH                %{install_path}/bin
+prepend-path    PYTHONPATH          %{install_path}/lib/python2.7/site-packages
+prepend-path    INCLUDE             %{install_path}/include
+prepend-path    LD_LIBRARY_PATH     %{install_path}/lib
+prepend-path    MANPATH             %{install_path}/share/man
+
+setenv          %{PNAME}_DIR        %{install_path}
+setenv          %{PNAME}_BIN        %{install_path}/bin
+setenv          %{PNAME}_LIB        %{install_path}/lib
+setenv          %{PNAME}_INC        %{install_path}/include
+
+EOF
+
+%files
+%defattr(-,root,root,-)
+%{OHPC_PUB}
+%doc README COPYING VERSION
+
+%changelog
+* Fri Jan 12 2018 Christopher M. Cantalupo <christopher.m.cantalupo@intel.com> v0.4.0
```

Comment:
@koomie I think we also should have a rule about the %changelog section. Most of our SPEC files have no %changelog at all. I would prefer to have a changelog, but as all is documented in git it is not really necessary.

If we have a changelog it should, however, only contain the changes to the SPEC file and not about the software packaged. This %changelog, from my point of view, is much too long and detailed, especially for a new OpenHPC package.

Reply:
Agreed, the %changelog in the spec file is intended to track changes to the spec file itself, not the software it package.  Appending the geopm/ChangeLog to the spec file is not appropriate here.  We can start a log for changes to the OpenHPC spec file and remove what is there.

Reply:
Done. Will be in next pull request rev.